### PR TITLE
WASP-1444\Add support DSPM Data Detection configuration for terraform provider

### PR DESCRIPTION
**Pull request by koby-orca**
### New resources & data sources
- **`orcasecurity_dspm_policy`** — manage DSPM scanning policies (detector/category/region/tag selectors, tags, advanced settings).
- **`orcasecurity_data_detection_rule`** — manage data detection rules (feature, action, priority, cloud-account/business-unit/tag selectors, attached policies).
- **`orcasecurity_sensitive_data_identifier`** (resource) — manage custom DSPM detectors (conditions, detection types, sensitivity, keywords, thresholds).
- **`orcasecurity_sensitive_data_identifiers`** (data source) — look up built-in and custom detectors, filterable by title/category.

### API client
- New entities: `dspm_detector`, `dspm_policy`, `data_detection_rule`, each with full CRUD and unit tests.
- Rule create uses `PUT` on the collection; updates go through `POST /bulk_rules` (partial — every mutable list must always be serialized). List endpoint returns a bare array (no `{status,data}` envelope). Policy `tags` may come back as `{}` or `["..."]`.

### Docs & examples
- Generated docs + `*.md.tmpl` templates and `examples/` for all new resources/data sources (via `go generate`; templates are the source of truth).

### Fixes rolled in
- **Test cognitive complexity (SonarQube):** extracted request/response assertions out of `RoundTripFunc` closures into top-level helpers so branches no longer incur the nesting penalty. Four DSPM API-client tests dropped from 17–28 down to ≤15.
- **`orcasecurity_shift_left_project` plan drift (pre-existing bug, confirmed against the live API):**
  - The API never echoes the write-only `default_policies` flag, so `Read` always saw `false` → perpetual `false -> true` diff. Now derived from whether any built-in policy is attached.
  - `support_*_via_cli` receive server-assigned defaults (`ALLOW`/`BLOCK`) that `Read` wrote back over a null config → `-> null` drift. Made `Optional + Computed` and populated from the create response.
  - The update endpoint ignores `default_policies` and rejects requests with no active policy; `Update` now preserves the currently-attached policy ids when policies are managed via `default_policies`.
- **`orcasecurity_webhook` data source acceptance test:** now provisions the webhook it looks up (and lets the test framework destroy it) instead of depending on a pre-existing org fixture.
- **`orcasecurity_discovery_view` description:** the Orca UI stores a view description under `extra_params.description` (confirmed against the live API), but the provider had no way to set or read it. Added an Optional `description` attribute that maps to `extra_params.description` and round-trips on Read/import.

- **`orcasecurity_data_detection_rule` requires a policy:** the Orca UI mandates at least one policy when creating a rule, but the API does not enforce it (rules with `policies:[]` create fine). Made `policies` Required with a min-1 validator so the provider matches the UI.

### Verification
- Unit tests pass; `go vet` clean.
- Live acceptance tests pass for the DSPM packages, the fixed `shift_left_project` (create → import → update, auto-destroyed), the webhook data source, `discovery_view` description (create → import round-trip → update), and `data_detection_rule` (Basic incl. do_not_scan update, ScopeRequired, PoliciesRequired). Behaviors confirmed directly against the live API; verified no leaked resources in the test org.

### CODE REVIEW

File: orcasecurity/api_client/data_detection_rule.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	if response.Data.ID == "" {
+		return nil, fmt.Errorf("rule retrieve: could not decode response: %s", string(resp.Body()))
+	}
+	return &response.Data, nil
+}
+
+// ListDataDetectionRules lists all scan configuration rules.
+// NOTE: the endpoint returns a bare JSON array, not the {status,data} envelope.
+func (client *APIClient) ListDataDetectionRules() ([]DataDetectionRule, error) {
```
Comment (aliakseis-orca) [inline orcasecurity/api_client/data_detection_rule.go]: i see it is unused, should we remove it?

File: examples/resources/orcasecurity_data_detection_rule/import.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+terraform import orcasecurity_sensitive_data_identifier.example IDENTIFIER_ID
```
Comment (aliakseis-orca) [inline examples/resources/orcasecurity_data_detection_rule/import.sh]: Wrong resource type — copy-pasted from the identifier example. Should be `terraform import orcasecurity_data_detection_rule.example RULE_ID`. This flows into the generated `docs/resources/data_detection_rule.md`, so re-run `go generate` after fixing.

File: examples/resources/orcasecurity_dspm_policy/import.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+terraform import orcasecurity_sensitive_data_identifier.example IDENTIFIER_ID
```
Comment (aliakseis-orca) [inline examples/resources/orcasecurity_dspm_policy/import.sh]: Wrong resource type — copy-pasted from the identifier example. Should be `terraform import orcasecurity_dspm_policy.example POLICY_ID`. This flows into the generated `docs/resources/dspm_policy.md`, so re-run `go generate` after fixing.

**Review by aliakseis-orca — CHANGES_REQUESTED** [review]

**Comment by koby-orca** [comment]
Addressed all review findings in 57d78b7:

**Finding 1 (blocker)** — All six `document` selectors in `orcasecurity_dspm_policy` are now `schema.SetAttribute` (with `setvalidator.SizeAtLeast(1)` on `detectors`). Also switched `policies` / `selector_cloud_accounts` / `selector_business_units` in `orcasecurity_data_detection_rule` to Sets in the same commit for consistency — they are conceptually sets and nothing is released yet, so no state migration. Top-level policy `tags` stay a List (server round-trips order as-is, per your live check). Nested rule `tags` selectors stay ListNested.

**Finding 2** — The duplicate-document 400 (`Policy with the same document already exists`) is now in the resource Description and the generated docs.

**Finding 3** — `tfconv` has table tests now; they caught a real edge case: a configured-empty list flipped to null in state when the API returned a nil slice (`ListValueFrom(nil)` produces a null list) — fixed in both PreserveNull helpers. Added `shift_left_project` tests for `hasBuiltinPolicy` / `GetShiftLeftProject` reconstruction, extracted the Update policy-id preservation into `attachedPolicyIDs` with its own unit test. dspm_policy acceptance test has a multi-element unsorted-document step plus a same-config `PlanOnly` step pinning the empty plan. Detector PUT full-replacement contract is pinned: `assertUpdateDetectorOmitsUnsetProperties` asserts every unset optional property is absent from the payload.

**Finding 4** — `assertMethodPath` is used across all new api_client tests. Envelope decoding is one generic `readData[T]` (api_client/envelope.go); missing/null `data` key is an error everywhere, and the policy/detector getters additionally reject a decoded entity with no id — the silent zero-value path is gone.

**On the `current != nil` guard** — verified you're right: `doRequest` returns a non-nil error for every non-OK response, so the `!resp.IsOk()` branch in `GetShiftLeftProject` was unreachable and `(nil, nil)` never happens. Dropped both the dead branch and the guard, documented the error-or-value contract on the function, and pinned it with a test (`TestGetShiftLeftProject_NotFoundIsError`).

Live acceptance runs: `dspm_policy`, `data_detection_rule`, `sensitive_data_identifier`, `shift_left_project` — all pass, including the new multi-element + empty-plan steps.

**Review by aliakseis-orca — APPROVED** [review]
